contrib/demo/testDemos.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
```

Comment:
This feels like an uncomfortable amount of Bash engineering for me. I _love_ that we'll have guards against demo regressions in CI, but I think we should consider some e2e tests in Go in the future.

Reply:
+1 for this in the future.
Though being able to use, in tests, the exact same scenarios as those we define in demos we show seemed relevant to me.

Reply:
I reworked this script be also be an easy and clean way to run a demo (one of the 2 demos that already exist, or additional demos in the future) either interactively or as an automatic test that can be triggered in a GH action.
Of course it's completely clear that GO-based E2E tests are still mandatory.